### ISSUE TITLE
fatal: [test_server]: FAILED! => {"changed": false, "msg": "Failed to import docker or docker-py - No module named 'docker'. Try `pip install docker` or `pip install docker-py` (Python 2.6)"}

### ISSUE THREAD

**Issue by ghost** [nicholasamorim/ansible-role-harbor#7]
It seems some things are missing if you don't enable the "behind proxy" flag but on my way to figure out what it is I am reporting more issues.
On my last run I got `fatal: [test_server]: FAILED! => {"changed": false, "msg": "Failed to import docker or docker-py - No module named 'docker'. Try `pip install docker` or `pip install docker-py` (Python 2.6)"}`

I think the role should include this to solve this error:
```
- name: Install base packages needed for docker install
  apt:
    name: "{{ packages }}"
    update_cache: yes
    state: latest
  vars:
    packages:
      - python3-pip
  become: true

- name: Install latest docker python bindings
  pip:
    name: "{{ packages }}"
    executable: pip3
    state: latest
  vars:
    packages:
      - docker
      - jsondiff ##check if needed, I know it is for swarm but don't know about compose
      - pyyaml ##check if needed, I know it is for swarm but don't know about compose
      - docker-compose
  become: true
```

**Comment by ghost** [comment]
What do you think?

**Comment by nicholasamorim** [comment]
Technically, installing docker and docker-compose is beyond the scope of this role.

But I'm all for making things easy, not a purist. I'll review the proposed change and most likely will add it, yes. I'll just make sure that is documented and that people can bypass it easily.

Again, thanks!

**Comment by ghost** [comment]
Those are the python API bindings

**Comment by nicholasamorim** [comment]
Ah right, I misread it. Then yeah, it makes sense the role can run on its own. Will add that change on Monday then.

**Comment by ghost** [comment]
My bad, this part is not,

```
- name: Install base packages needed for docker install
  apt:
    name: "{{ packages }}"
    update_cache: yes
    state: latest
  vars:
    packages:
      - python3-pip
  become: true
```
 this is
```
- name: Install latest docker python bindings
  pip:
    name: "{{ packages }}"
    executable: pip3
    state: latest
  vars:
    packages:
      - docker
      - jsondiff ##check if needed, I know it is for swarm but don't know about compose
      - pyyaml ##check if needed, I know it is for swarm but don't know about compose
      - docker-compose
  become: true
```

**Comment by nicholasamorim** [comment]
I've solved this, however:

- I removed the `executable` and leave that to be the Python/Pip that the ansible connection itself is using.
- Removed `jsondiff`.
- Changed `latest` to `present`. If the user has the bindings already installed, I don't think it's wise for this role to go and update them, it might break something for the user.
- Added version identifiers to `pyyaml` (>=3.11) and `docker-compose` (>=1.7.0) as per the Ansible docs.

Thanks!
